gson/src/main/java/com/google/gson/stream/JsonWriter.java

```diff
@@ -225,6 +230,24 @@ public final void setIndent(String indent) {
     }
   }
 
+  /**
+   * Sets the newline string to be used in the encoded document.
+   *
+   * <p>Has no effect if the serialized format is a single line.</p>
+   *
+   * @param newline the string that will be used for newline.
+   */
+  public final void setNewline(String newline) {
+    this.newline = Objects.requireNonNull(newline, "newline == null");
+  }
+
+  /**
+   * Returns the newline string used by this writer.
+   */
+  public final String getNewline() {
```

Comment:
Would probably be good to add `@since $next-version$` here as well.

Reply:
Done.